src/tree_builder/rules.rs

```diff
@@ -1179,18 +1193,60 @@ impl<Handle, Sink> TreeBuilderStep
             }),
 
             //§ parsing-main-intemplate
-            InTemplate => {
-                // NB: Implementing <template> requires not just adding
-                // the InTemplate rules here, but also inserting various
-                // extra logic at other points noted throughout the
-                // parser.
-
-                if self.opts.ignore_missing_rules {
-                    self.step(InBody, token)
-                } else {
-                    panic!("FIXME: <template> not implemented");
+            InTemplate => match_token!(token {
+                CharacterTokens(_, _) => self.step(InBody, token),
+                CommentToken(_) => self.step(InBody, token),
```

Comment:
`DocTypeToken` is already implicitly rejected in all insertion modes (see for example the BeforeHtml branch), I just continued that trend.